isotovideo

```diff
@@ -386,133 +395,185 @@ sub check_asserted_screen {
+    my ($io_handle, $rsp) = @_;
+
+    if ($rsp->{cmd} =~ m/^backend_(.*)/) {
+        return if handle_paused_test_execution($io_handle, $rsp);
+        die 'isotovideo: we need to implement a backend queue' if $backend_requester;
+        $backend_requester = $io_handle;
+        my $cmd = $1;
+        delete $rsp->{cmd};
+
+        # send debugging info for ws clients
+        myjsonrpc::send_json($cmd_srv_fd, {$cmd => $rsp});
+        # pass command to backend
+        myjsonrpc::send_json($bmwqemu::backend->{to_child}, {cmd => $cmd, arguments => $rsp});
+        return;
+    }
+    if ($rsp->{cmd} eq 'set_pause_at_test') {
+        $pause_test_name = $rsp->{name};
+
+        # print/send debugging info
+        print("isotovideo: test execution will be paused at test $pause_test_name\n");
+        myjsonrpc::send_json($cmd_srv_fd, {set_pause_at_test => $pause_test_name});
+
+        myjsonrpc::send_json($io_handle, {ret => 1});
+        return;
+    }
+    if ($rsp->{cmd} eq 'resume_test_execution') {
+        # print/send debug info
+        print($stop_passing_commands_to_backend ?
+              "isotovideo: test execution will be resumed\n"
+            : "isotovideo: resuming test execution requested but not paused anyways\n"
+        );
+        myjsonrpc::send_json($cmd_srv_fd, {resume_test_execution => $postponed_command});
+
+        # no longer stop passing commands to backend
+        $stop_passing_commands_to_backend = 0;
+
+        # resume with postponed command so autotest can continue
+        if ($postponed_command) {
+            my $cmd = $postponed_command->{cmd};
+            print("isotovideo: resuming, continue passing $cmd to backend\n");
+            process_command($postponed_io_handle, $postponed_command);
+            $postponed_io_handle = undef;
+            $postponed_command   = undef;
+            return;
         }
-        if ($rsp->{cmd} eq 'check_screen') {
-            $no_wait = $rsp->{no_wait} // 0;
-
-            $tags = $bmwqemu::backend->_send_json(
-                {
-                    cmd       => 'set_tags_to_assert',
-                    arguments => {
-                        mustmatch => $rsp->{mustmatch},
-                        timeout   => $rsp->{timeout},
-                        check     => $rsp->{check}}})->{tags};
-            next;
+        myjsonrpc::send_json($io_handle, {ret => 1});
+        return;
+    }
+    if ($rsp->{cmd} eq 'set_current_test') {
+        $bmwqemu::backend->_send_json({cmd => 'set_serial_offset'});
+        $current_test_name      = $rsp->{name};
+        $current_test_full_name = $rsp->{full_name};
+
+        # send debugging info for ws clients
+        myjsonrpc::send_json($cmd_srv_fd, {set_current_test => $current_test_name});
+        if (   $pause_test_name
+            && $current_test_name
+            && $current_test_full_name
+            && ($pause_test_name eq $current_test_name || $pause_test_name eq $current_test_full_name))
```

Comment:
Why `$current_test_name` and `$current_test_full_name`? what's the difference between both?